### PR TITLE
xds: clean up some ImmutableMaps

### PR DESCRIPTION
**Pull request by dapengzhang0**
Change
```java
          ImmutableMap.Builder<XdsLocality, LocalityLbInfo> builder = ImmutableMap.builder();
          for (Map.Entry<XdsLocality, LocalityLbInfo> entry : localityMap.entrySet()) {
            if (!entry.getKey().equals(locality)) {
              builder.put(entry);
            }
          }
          localityMap = builder.build();
```

into
```java
          Map<XdsLocality, LocalityLbInfo> copy = new LinkedHashMap<>(localityMap);
          copy.remove(locality);
          localityMap = ImmutableMap.copyOf(copy);
```

The performance is almost the same, unless we use the beta API `ImmutableMap.builderWithExpectedSize()`. But the latter has much better readability.

### CODE REVIEW

File: xds/src/main/java/io/grpc/xds/XdsComms.java [hunk 1]
```diff
@@ -239,7 +238,8 @@ public void run() {
                   localityStore.updateDropPercentage(dropOverloads);
 
                   List<LocalityLbEndpoints> localities = clusterLoadAssignment.getEndpointsList();
-                  Map<XdsLocality, LocalityInfo> localityEndpointsMapping = new LinkedHashMap<>();
+                  ImmutableMap.Builder<XdsLocality, LocalityInfo> localityEndpointsMappingBuilder =
```
Comment (voidzcy) [inline xds/src/main/java/io/grpc/xds/XdsComms.java]: Why do this inverse change?

File: xds/src/main/java/io/grpc/xds/LocalityStore.java [hunk 2]
```diff
@@ -306,13 +307,9 @@ private void deactivate(final XdsLocality locality) {
...
           localityLbInfo.shutdown();
 
-          ImmutableMap.Builder<XdsLocality, LocalityLbInfo> builder = ImmutableMap.builder();
-          for (Map.Entry<XdsLocality, LocalityLbInfo> entry : localityMap.entrySet()) {
-            if (!entry.getKey().equals(locality)) {
-              builder.put(entry);
-            }
-          }
-          localityMap = builder.build();
+          Map<XdsLocality, LocalityLbInfo> copy = new LinkedHashMap<>(localityMap);
```
Comment (voidzcy) [inline xds/src/main/java/io/grpc/xds/LocalityStore.java]: You can just make `localityMap` itself a `LinkedHashMap`, as we discussed in https://github.com/grpc/grpc-java/pull/6110#discussion_r321533255. It doesn't have to be immutable.

File: xds/src/main/java/io/grpc/xds/XdsComms.java (hunk 1 above)
Comment (voidzcy) [reply xds/src/main/java/io/grpc/xds/XdsComms.java]: You are getting rid of `ImmutableMap.Builder` pattern. But this change changes from `Collections.unmodifiableMap(...)` to using `ImmutableMap.Builder`.

File: xds/src/main/java/io/grpc/xds/LocalityStore.java (hunk 2 above)
Comment (dapengzhang0) [reply xds/src/main/java/io/grpc/xds/LocalityStore.java]: PTALA

File: xds/src/main/java/io/grpc/xds/XdsComms.java (hunk 1 above)
Comment (dapengzhang0) [reply xds/src/main/java/io/grpc/xds/XdsComms.java]: Did I? I don't see it was ever `ImmutableMap.Builder` before. https://github.com/grpc/grpc-java/commit/213b91b16583b0a982c8c32efbbb3f2a2406356c#diff-9e293ee2079eae78a4ffd73aeb825704R221

**Pull request merged**